### PR TITLE
Deprecate pre-compiled eBPF on certain platforms

### PR DESCRIPTION
**Pull request by hmahmood**
### What does this PR do?
Deprecates the prebuilt variants of the network tracer module (NPM), USM, and eBPF conntracker (NPM) on:
1. RHEL kernel versions >= 5.14
2. non-RHEL kernel versions >= 6

#### `system_probe_config.allow_precompiled_fallback` Key Deprecation
The `system_probe_config.allow_precompiled_fallback` has been deprecated in favor of `system_probe_config.allow_prebuilt_fallback`. This aligns with references in code as well logs. In addition, the default value for `system_probe_config.allow_precompiled_fallback` has been removed as the default value makes it impossible to tell if the key was set in the config or not; this is required to deprecate the key correctly.

#### Affects on Network Tracer, USM Monitor, and eBPF Conntracker Loading
With this change if CO-RE or runtime-compiled variants of the network tracer fail to load, the prebuilt network tracer/USM monitor/eBPF conntracker will not load on kernel versions >= 6 or RHEL kernel versions >= 5.14, and system-probe will fail to load these components.

The prebuilt versions of these components can still load under these circumstances:
1. both CO-RE and runtime compiled variants are disabled
2. `system_probe_config.allow_prebuilt_fallback` is set to `true` in the config

When the prebuilt variants are loaded, we emit a warning in the logs on platforms where it is deprecated.

For kernels < 6 and RHEL kernels < 5.14, the status quo will remain, i.e. prebuilt fallback will be enabled (if not disabled explicitly in the config), and no warning will be emitted if the prebuilt variant is loaded.

#### KMT Tests for Prebuilt
This PR also disables running the prebuilt versions of the kmt tests for the network tracer, USM, and eBPF conntracker on the deprecated platforms.

### CODE REVIEW

File: releasenotes/notes/deprecate-precompiled-ebpf-d887926d6a5fe377.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+---
+deprecations:
+  - |
+    Precompiled eBPF for the network tracer system-probe module has been
+    deprecated in favor of CO-RE and runtime compilation variants on Linux
+    kernel versions 6+ and RHEL kernel versions 5.14+. To continue to use
+    the precompiled eBPF network tracer, set
+    `system_probe_config.allow_precompiled_fallback` in the
+    system-probe config file, or set the environment variable
+    `DD_ALLOW_PRECOMPILED_FALLBACK` to `true` on these platforms.
```
Comment (maycmlee) [inline releasenotes/notes/deprecate-precompiled-ebpf-d887926d6a5fe377.yaml:17]:
```suggestion
    `DD_ALLOW_PRECOMPILED_FALLBACK` to `true` on those platforms.
```

**Review by maycmlee — APPROVED** [review]
Just a small nit, but approving.

**Review by dustmop — APPROVED** [review]
LGTM for ASC-owned files

File: pkg/ebpf/precompiled/deprecate.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux
+
+// Package precompiled implements precompiled specific eBPF functionality
+package precompiled
```
Comment (brycekahle) [inline pkg/ebpf/precompiled/deprecate.go:9]: I realize the config uses the term `precompiled`, but we use the term prebuilt pretty much everywhere else. I'd like to be consistent and use `prebuilt` where possible.

Comment (hmahmood) [reply pkg/ebpf/precompiled/deprecate.go:9]: Even in the log messages? I have no preference either way.

Comment (brycekahle) [reply pkg/ebpf/precompiled/deprecate.go:9]: Yeah. WDYT about deprecating the `precompiled` config value and adding one using `prebuilt`?

Comment (hmahmood) [reply pkg/ebpf/precompiled/deprecate.go:9]: We would have to still get rid of the `precompiled` key default for the deprecation to work correctly I think, but I need to verify this.

Comment (brycekahle) [reply pkg/ebpf/precompiled/deprecate.go:9]: Correct, you cannot have a default if you need to detect if the value is set or not.

Comment (hmahmood) [reply pkg/ebpf/precompiled/deprecate.go]: Done, please verify that the new test in `config_linux_test.go` does the right thing.

**Comment by hmahmood** [comment]
@dustmop can I bother you to review this again since we changed the name of the config key?

**Review by dustmop — APPROVED** [review]
Approving again

File: cmd/system-probe/config/config_linux.go [hunk 3]
```diff
@@ -8,10 +8,17 @@
...
 import (
+	"github.com/DataDog/datadog-agent/pkg/config/model"
+	"github.com/DataDog/datadog-agent/pkg/ebpf/prebuilt"
 	ebpfkernel "github.com/DataDog/datadog-agent/pkg/security/ebpf/kernel"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
+var (
+	allowPrebuiltFallbackKey    = spNS("allow_prebuilt_fallback")
+	allowPrecompiledFallbackKey = spNS("allow_precompiled_fallback")
```
Comment (brycekahle) [inline cmd/system-probe/config/config_linux.go:19]: I think this is unused now?

**Review by brycekahle — APPROVED** [review]

**Review by akarpz — APPROVED** [review]

File: pkg/network/usm/ebpf_main.go [hunk 4]
```diff
@@ -201,12 +201,16 @@ func (e *ebpfProgram) Init() error {
...
 
-		if !e.cfg.AllowPrecompiledFallback {
+		if !e.cfg.AllowPrebuiltFallback {
 			return fmt.Errorf("runtime compilation failed: %w", err)
 		}
 		log.Warnf("runtime compilation failed: attempting fallback: %s", err)
 	}
 
+	if prebuilt.IsDeprecated() {
+		log.Warnf("using deprecated prebuilt USM monitor")
```
Comment (guyarb) [inline pkg/network/usm/ebpf_main.go:211]:
```suggestion
		log.Warn("using deprecated prebuilt USM monitor")
```

**Review by guyarb — APPROVED** [review]

**Pull request merged**